thor/src/protocols/splice.rs

```diff
@@ -147,31 +153,47 @@ impl State0 {
         Message0 {
             R: R_other,
             Y: Y_other,
-            splice_in: splice_in_other,
+            splice: splice_other,
         }: Message0,
     ) -> Result<State1> {
-        let previous_funding_txin = self.previous_tx_f.as_txin();
-        let previous_funding_psbt = PartiallySignedTransaction::from_unsigned_tx(Transaction {
-            version: 2,
-            lock_time: 0,
-            input: vec![previous_funding_txin],
-            output: vec![],
-        })
-        .expect("Only fails if script_sig or witness is empty which is not the case.");
-
         let mut our_balance = self.previous_balance.ours;
         let mut their_balance = self.previous_balance.theirs;
 
+        let mut splice_outputs = vec![];
         let mut splice_in_inputs = vec![];
 
-        if let Some(splice_in) = self.splice_in_self.clone() {
-            splice_in_inputs.push(splice_in.input_psbt);
-            our_balance += splice_in.amount;
+        match self.splice_self.clone() {
+            Splice::Out(tx_out) => {
+                if tx_out.value > self.previous_balance.ours.as_sat() {
+                    anyhow::bail!("We are splicing out more than we have");
+                } else {
```

Comment:
Don't need this else after `bail!`. Also please consider adding a use statement for `bail`, we have `use anyhow::{Context, Result};` already in this file (on master, was just added recently :)